lib/charms/layer/basic.py

```diff
@@ -199,7 +199,13 @@ def bootstrap_charm_deps():
         # a set so that we can ignore the pre-install packages and let pip
         # choose the best version in case there are multiple from layer
         # conflicts)
-        pkgs = _load_wheelhouse_versions().keys() - set(pre_install_pkgs)
+        _versions = _load_wheelhouse_versions()
+        _pkgs = _versions.keys() - set(pre_install_pkgs)
+        # add back the versions such that each package in pkgs is
+        # <package_name>==<version>.
+        # This ensures that pip 20.3.4+ will install the packages from the
+        # wheelhouse without (erroneously) flagging an error.
+        pkgs = _add_back_versions(_pkgs, _versions)
```

Comment:
So apparently if I understand correctly this is not only adding the version back but also removing extensions?

Reply:
Yes; you can't ask pip install <package>==<version>[.tar.gz|.zip] as it will throw an error.  Unfortunately, `LooseVersion()` wrapping the version also includes the extension and so it needs to be removed.